tests/tests_pytorch/strategies/test_colossalai_strategy.py

```diff
@@ -0,0 +1,291 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+import pytest
+import torch
+import torch.nn.functional as F
+from torch import nn, Tensor
+from torch.optim import Optimizer
+from torchmetrics import Accuracy
+
+from pytorch_lightning import LightningModule, seed_everything, Trainer
+from pytorch_lightning.callbacks import ModelCheckpoint
+from pytorch_lightning.demos.boring_classes import BoringModel
+from pytorch_lightning.plugins.precision import ColossalAIPrecisionPlugin
+from pytorch_lightning.strategies import ColossalAIStrategy
+from pytorch_lightning.strategies.colossalai import _COLOSSALAI_AVAILABLE
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from tests_pytorch.helpers.datamodules import ClassifDataModule
+from tests_pytorch.helpers.runif import RunIf
+
+if _COLOSSALAI_AVAILABLE:
+    from colossalai.nn.optimizer import HybridAdam
+
+
+def test_invalid_colosalai(monkeypatch):
+    import pytorch_lightning.strategies.colossalai as colossal_strategy
+
+    monkeypatch.setattr(colossal_strategy, "_COLOSSALAI_AVAILABLE", False)
+    with pytest.raises(
+        MisconfigurationException,
+        match="To use the `ColossalAIStrategy`, please install `colossalai` first. "
+        "Download `colossalai` by consulting `https://colossalai.org/download`.",
+    ):
+        ColossalAIStrategy()
+
+
+@RunIf(colossalai=True)
+def test_colossalai_strategy_with_trainer_by_instance(tmpdir):
+    trainer = Trainer(fast_dev_run=True, default_root_dir=tmpdir, precision=16, strategy=ColossalAIStrategy())
```

Comment:
Trainer doesn't start

```suggestion
def test_colossalai_strategy_with_trainer_by_instance():
    trainer = Trainer(precision=16, strategy=ColossalAIStrategy())
```